addons/im_livechat/static/src/embed/common/thread_model_patch.js

```diff
@@ -6,6 +6,16 @@ import { patch } from "@web/core/utils/patch";
 import { SESSION_STATE } from "./livechat_service";
 import { _t } from "@web/core/l10n/translation";
 
+/** @type {typeof Thread} */
+const threadStaticPatch = {
+    async getOrFetch() {
+        // wait for restore of livechatService.savedState as channel might be inserted from there
+        await this.store.isReady;
+        return super.getOrFetch(...arguments);
+    },
```

Comment:
Note: hack for when channel id = -1 (non-persisted).
Standard `getOrFetch` won't be able to find the channel from the server, but this info might be inside the `savedState`.
Wait for restore of `savedState` before discarding the channel as non-existing.